runners/jstorm/src/main/java/org/apache/beam/runners/jstorm/JStormRunnerResult.java

```diff
@@ -17,47 +17,92 @@
-import org.apache.beam.sdk.transforms.Aggregator;
 import org.joda.time.Duration;
 
 /**
  * A {@link PipelineResult} of executing {@link org.apache.beam.sdk.Pipeline Pipelines} using
  * {@link JStormRunner}.
  */
-public class JStormRunnerResult implements PipelineResult {
-    @Override
-    public State getState() {
-        throw new UnsupportedOperationException("This method is not yet supported.");
+public abstract class JStormRunnerResult implements PipelineResult {
+
+  public static JStormRunnerResult local(
+      String topologyName,
+      Config config,
+      LocalCluster localCluster,
+      long localModeExecuteTimeSecs) {
+    return new LocalStormPipelineResult(
+        topologyName, config, localCluster, localModeExecuteTimeSecs);
+  }
+
+  private final String topologyName;
+  private final Config config;
+
+  JStormRunnerResult(String topologyName, Config config) {
+    this.config = checkNotNull(config, "config");
+    this.topologyName = checkNotNull(topologyName, "topologyName");
+  }
+
+  public State getState() {
+    return null;
+  }
+
+  public Config getConfig() {
+    return config;
+  }
+
+  public String getTopologyName() {
+    return topologyName;
+  }
+
+  private static class LocalStormPipelineResult extends JStormRunnerResult {
+
+    private LocalCluster localCluster;
+    private long localModeExecuteTimeSecs;
+
+    LocalStormPipelineResult(
+        String topologyName,
+        Config config,
+        LocalCluster localCluster,
+        long localModeExecuteTimeSecs) {
+      super(topologyName, config);
+      this.localCluster = checkNotNull(localCluster, "localCluster");
     }
 
     @Override
     public State cancel() throws IOException {
-        throw new UnsupportedOperationException("This method is not yet supported.");
+      //localCluster.deactivate(getTopologyName());
+      localCluster.killTopology(getTopologyName());
+      localCluster.shutdown();
+      JStormUtils.sleepMs(1000);
+      return State.CANCELLED;
     }
 
     @Override
     public State waitUntilFinish(Duration duration) {
-        throw new UnsupportedOperationException("This method is not yet supported.");
+      return waitUntilFinish();
     }
 
     @Override
     public State waitUntilFinish() {
-        throw new UnsupportedOperationException("This method is not yet supported.");
-    }
-
-    @Override
-    public <T> AggregatorValues<T> getAggregatorValues(Aggregator<?, T> aggregator)
-            throws AggregatorRetrievalException {
-        throw new UnsupportedOperationException("This method is not yet supported.");
+      JStormUtils.sleepMs(localModeExecuteTimeSecs * 1000);
+      try {
+        return cancel();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
     }
 
     @Override
     public MetricResults metrics() {
-        throw new UnsupportedOperationException("This method is not yet supported.");
+      return null;
```

Comment:
<!--new_thread; commit:dec02cd228da604a68e5779a5c10935881fd37a1; resolved:0-->
Should still throw, yes? Still unsupported, clearly.